lib/repl.js

```diff
@@ -496,6 +506,44 @@ function REPLServer(prompt,
     self.displayPrompt(true);
   });
 
+  // Wrap readline tty to enable editor mode
+  const ttyWrite = self._ttyWrite.bind(self);
+  self._ttyWrite = (d, key) => {
+    if (!self.editorMode || !self.terminal) {
+      ttyWrite(d, key);
+      return;
+    }
+
+    // editor mode
+    if (key.ctrl && !key.shift) {
+      switch (key.name) {
+        case 'd': // End editor mode
+          self.turnOffEditorMode();
+          sawCtrlD = true;
+          ttyWrite(d, { name: 'return' });
+          break;
+        case 'n': // Override next history item
+        case 'p': // Override previous history item
+          break;
+        case 'z':
+          if (process.platform !== 'win32') {
+            self.turnOffEditorMode();
```

Comment:
This is not documented. Why Ctrl Z turns editor mode off on other platforms? And why it doesn't on Windows alone?